app/src/main/java/com/besome/sketch/ProjectsFragment.java

```diff
@@ -352,10 +375,39 @@ public int a() {
             return projectsList.size();
         }
 
+        private boolean matchesQuery( HashMap<String, Object> projectMap){
+            final String searchQuery = projectsSearchView.getQuery().toString().toLowerCase();
+            if(searchQuery.isEmpty()) return true;
+
+            String scId = yB.c(projectMap, "sc_id").toLowerCase();
+            if(searchQuery.contains(scId) || scId.contains(searchQuery)) return true;
+
+            String appName = yB.c(projectMap, "my_ws_name").toLowerCase();
+            if(searchQuery.contains(appName) || appName.contains(searchQuery)) return true;
+
+            String projectName = yB.c(projectMap, "my_app_name").toLowerCase();
+            if(searchQuery.contains(projectName) || projectName.contains(searchQuery)) return true;
+
+            String packageName = yB.c(projectMap, "my_sc_pkg_name").toLowerCase();
+            return searchQuery.contains(packageName) || packageName.contains(searchQuery);
+        }
+
         @Override
         public void b(ViewHolder viewHolder, int position) {
             HashMap<String, Object> projectMap = projectsList.get(position);
             String scId = yB.c(projectMap, "sc_id");
+
+            //Show hide based on search query
+            if (matchesQuery(projectMap)) {
+                viewHolder.b.setVisibility(View.VISIBLE);
+                viewHolder.b.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
+            } else {
+                viewHolder.b.setVisibility(View.GONE);
+                viewHolder.b.setLayoutParams(new ViewGroup.LayoutParams(0, 0));
+
+            }
```

Comment:
This is a no-go for me. Please make the RecyclerView actually recycle items well.